youtube_dl/postprocessor/embedthumbnail.py

```diff
@@ -40,21 +40,51 @@ def run(self, info):
                 'Skipping embedding the thumbnail because the file is missing.')
             return [], info
 
+        if info['ext'] not in ['mp3', 'mkv', 'm4a', 'mp4']:
+        	raise EmbedThumbnailPPError('Only mp3, mkv and m4a/mp4 are supported for thumbnail embedding for now.')
+
         if info['ext'] == 'mp3':
             options = [
                 '-c', 'copy', '-map', '0', '-map', '1',
                 '-metadata:s:v', 'title="Album cover"', '-metadata:s:v', 'comment="Cover (Front)"']
+            input_paths = [filename, thumbnail_filename]
+
+        elif info['ext'] == 'mkv':
+            if thumbnail_filename.endswith(('.jpe', '.jpeg', '.jpg', '.jfif')):
+                mimetype = 'image/jpeg'
+                extension = 'jpg'
+            elif thumbnail_filename.endswith('.png'):
+                mimetype = 'image/png'
+                extension = 'png'
+            else:
+                self._downloader.report_warning(
+                    'Skipping embedding the thumbnail because the thumbnail extension is unknown.')
+                return [], info
 
+            options = [
+                '-c', 'copy',
```

Comment:
```suggestion
                '-c', 'copy',
                '-map', '0',
```
`-map 0` is needed to copy all streams from the source file.